api4/webhook.go

```diff
@@ -451,33 +452,39 @@ func incomingWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 	r.ParseForm()
 
 	var payload io.Reader
+	incomingWebhookPayload := &model.IncomingWebhookRequest{}
 	contentType := r.Header.Get("Content-Type")
 	if strings.Split(contentType, "; ")[0] == "application/x-www-form-urlencoded" {
 		payload = strings.NewReader(r.FormValue("payload"))
+	} else if strings.Split(contentType, "; ")[0] == "multipart/form-data" {
+		r.ParseMultipartForm(0)
+
+		decoder := schema.NewDecoder()
+		err := decoder.Decode(incomingWebhookPayload, r.PostForm)
+
+		if err != nil {
+			c.Err = model.NewAppError("incomingWebhook", "api.webhook.incoming.error", nil, err.Error(), http.StatusBadRequest)
+			return
+		}
 	} else {
 		payload = r.Body
 	}
 
-	if c.App.Config().LogSettings.EnableWebhookDebugging {
-		var err error
-		payload, err = utils.InfoReader(
-			payload,
-			utils.T("api.webhook.incoming.debug"),
-		)
-		if err != nil {
-			c.Err = model.NewAppError("incomingWebhook", "api.webhook.incoming.debug.error", nil, err.Error(), http.StatusInternalServerError)
+	if strings.Split(contentType, "; ")[0] != "multipart/form-data" {
```

Comment:
Instead of checking this condition twice, I think it's probably better to just do this decoding right in the above if statement. I think duplicating (or throw it in a function if you don't want to duplicate) some code is better than splitting the flow into two different if statements and checking the same condition twice

Reply:
ok :)

Reply:
done